controller/src/controller.rs

```diff
@@ -203,3 +205,188 @@ impl<T: BottlerocketShadowClient> BrupopController<T> {
         }
     }
 }
+
+// Get node and BottlerocketShadow names
+#[instrument]
+fn get_associated_bottlerocketshadow_name() -> Result<String> {
+    let associated_node_name = env::var("MY_NODE_NAME").context(error::GetNodeName)?;
+    let associated_bottlerocketshadow_name = brs_name_from_node_name(&associated_node_name);
+
+    event!(
+        Level::INFO,
+        ?associated_bottlerocketshadow_name,
+        "Found associated bottlerocketshadow name."
+    );
+
+    Ok(associated_bottlerocketshadow_name)
+}
+
+/// sort shadows list which uses to determine the order of node update
+/// logic1: sort shadows by crash count
+/// logic2: move the shadow which associated bottlerocketshadow node hosts controller pod to the last
+#[instrument(skip())]
+fn sort_shadows(shadows: &mut Vec<BottlerocketShadow>, associated_brs_name: &str) -> Result<()> {
```

Comment:
nit: I think this function cannot return an error value. You can probably remove the return type.